src/App.tsx

```diff
@@ -14,37 +13,117 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+// Use this function in the render method to prepare goods
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  // Sort and reverse goods if needed
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((a, b) => (a > b ? 1 : -1));
+      break;
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+// DON'T save goods to the state
+// type State = {
+//   isReversed: boolean,
+//   sortType: SortType,
+// };
 
 export const App: React.FC = () => {
+  const [order, setOrder] = React.useState<ReorderOptions>({
+    sortType: SortType.NONE,
+    isReversed: false,
+  });
+
+  const alphabeticalOrder = () =>
+    setOrder({
+      sortType: SortType.ALPHABET,
+      isReversed: order.isReversed,
+    });
+
+  const orderByLength = () =>
```

Comment:
We can also simplify it here by having a single handler for ordering items, that will receive a sorting type as an argument. If your code and sorting types grows - you may just found yourself flooded with these similar-looking helpers.